### PR TITLE
Delete index before creation in test macro

### PR DESCRIPTION
**Pull request by bidoubiwa**
Partial fixes of #238

### CODE REVIEW

File: meilisearch-test-macro/src/lib.rs [hunk 1]
```diff
@@ -95,9 +95,13 @@ pub fn meilisearch_test(params: TokenStream, input: TokenStream) -> TokenStream
...
 
-        // And finally if an index was asked we create it and wait until meilisearch confirm its creation.
-        // We’ll need to delete it later.
+        // And finally if an index was asked we delete it, and we (re)create it and wait until meilisearch confirm its creation.
         if use_index {
+            outer_block.push(parse_quote!(client
+                .delete_index(&name)
+                .await
+                .unwrap();
+            ));
```
Comment (irevoire) [inline meilisearch-test-macro/src/lib.rs:104]:
```suggestion
            outer_block.push(parse_quote!({
                let res = client
                    .delete_index(&name)
                    .await
                    .unwrap()
                    .wait_for_completion(&client, None, None)
                    .await
                    .unwrap();
                if res.is_failure() {
                    let error = res.unwrap_failure();
                    assert_eq!(
                        error.error_code,
                        crate::errors::ErrorCode::IndexNotFound,
                        "{:?}",
                        error
                    );
                }
            }));

```

**Review by irevoire — APPROVED** [review]
Thanks for the contribution.
Could you keep the issue open since it was about two things and only one is half-solved? :grin:

**Comment by bidoubiwa** [comment]
bors merge

**Comment by bidoubiwa** [comment]
bors merge 😡

**Pull request merged**